### PR TITLE
Adds an option for "extended picks" + various new packages

### PR DESCRIPTION
**Pull request by MSoegtropIMC**
It also adds various new packages to 8.13~2021.09

### CODE REVIEW

File: shell_scripts/create_smoke_test_kit.sh [hunk 1]
```diff
@@ -248,7 +256,7 @@ echo 'ECHO "====================== SMOKE TEST SUCCESS ======================"'$'
 ##### Run bash runner script #####
 
 chmod u+x $smoke_script
-echo "On unix you can now run " $smoke_script
+echo "On macOS, linus or unix you can now run " $smoke_script
```
Comment (Zimmi48) [inline shell_scripts/create_smoke_test_kit.sh:259]:
```suggestion
echo "On macOS, Linux or unix you can now run " $smoke_script
```

**Comment by Zimmi48** [comment]
How do you decide what goes into the normal pick vs what goes into the extended pick?

**Comment by MSoegtropIMC** [comment]
> How do you decide what goes into the normal pick vs what goes into the extended pick?

the criteria for full are:

- the maintainer agreed to support the package
- the package does look well maintained (I decide this by the availability date of 8.13 and 8.12 compatible tags)
- no 0.X version number
- the build time is not excessive (in case I might add a separate question as for VST)

Up to now I didn't come across anything which I thought would be controversial. The only exclusion rule I applied so far is the 0.X rule. If I come across something which is controversial, I will discuss it in the PR. Of course this can also be changed.

File: linux/create_snapcraft_yaml.sh [hunk 2]
```diff
@@ -1,20 +1,40 @@
...
+# See https://creativecommons.org/publicdomain/zero/1.0/legalcode.txt
+
+###################### CREATE scnapcraft.yml #####################
+
+# This script creates snap craft yaml file from which a
+# snap based Linux installer is created in a later step
+
+###################### GATHER INFORMATION #####################
+
+# Run the Coq Platform setup scripts but stip just after opam install.
```
Comment (gares) [inline linux/create_snapcraft_yaml.sh:18]:
```suggestion
# Run the Coq Platform setup scripts but stop just after opam install.
```

File: coq_platform_make_windows.bat [hunk 3]
```diff
@@ -473,6 +473,7 @@ ECHO ========== BATCH FUNCTIONS ==========
   ECHO Except for expert users this is the recommended way to run this script.
   ECHO(
   ECHO -extent=f    Setup opam and build full Coq platform
+  ECHO -extent=f    Setup opam and build extended Coq platform
```
Comment (gares) [inline coq_platform_make_windows.bat:476]:
```suggestion
  ECHO -extent=x    Setup opam and build extended Coq platform

```

File: shell_scripts/init_utilities.sh [hunk 4]
```diff
@@ -112,6 +112,35 @@ function ask_user_opt3_cancel {
   done
 }
 
+# ------------------------------------------------------------------------------
+# Ask a four option + cancel question
+# Set environment variable ANSWER to the result
```
Comment (gares) [inline shell_scripts/init_utilities.sh:117]: an example of invokation would help

File: versions/packages-8.13.sh [hunk 5]
```diff
@@ -29,71 +29,122 @@ COQ_PLATFORM_USE_DEV_REPOSITORY='N'
...
 # - Packages with system dependencies should be given first.
 #   This avoids multiple sudo password requests
 # - Packages which take a long time to build should be given last.
 #   There is some evidence that they are built early then.
 
-PACKAGES="coq.8.13.2"
+########## BASE PACKAGES ##########
+
+# Build tools - this is selected early to avoid the version is changed later
+# and everything has to be recmpiled.
```
Comment (gares) [inline versions/packages-8.13.sh:43]:
```suggestion
# and everything has to be recompiled.
```

**Review by gares — APPROVED** [review]

**Pull request merged**